pkg/controller/kubedirectorcluster/members.go

```diff
@@ -1043,6 +1043,21 @@ func appConfig(
 	if setupErr != nil {
 		return true, setupErr
 	}
+	// Run the config file iff the event is registered during initial configuration.
+	appCr, appErr := catalog.GetApp(cr)
+	if appErr != nil {
+		shared.LogError(
+			reqLogger,
+			appErr,
+			cr,
+			shared.EventReasonCluster,
+			"app referenced by cluster does not exist",
+		)
+	}
+	role := catalog.GetRoleFromID(appCr, roleName)
+	if role.EventList != nil && !shared.StringInList("configure", *role.EventList) {
+		return false, appErr
```

Comment:
I'm curious about returning false here (for the "result is final" boolean). Previously you were returning true, which seemed correct. What caused the change?

Also you could/should return nil instead of appErr (for that second value) at this point.